Rover/mode.cpp

```diff
@@ -444,6 +448,38 @@ void Mode::set_steering(float steering_value)
     g2.motors.set_steering(steering_value);
 }
 
+// calculate target roll and pitch angles and output
+void Mode::calc_roll_pitch()
```

Comment:
I think we need to split this function up into two or three pieces.  So there should be:

1. method(s) in the mode class to transform the pilot's inputs into lean angles
2. method that takes target lean angles and passes them to the attitude controller and maybe then pushes them directly into the motors class.

Reply:
I have split into 2 methods and a helper that does them both, hopefully this it is closer to what you had in mind